### PR TITLE
polish docs

### PR DESCRIPTION
**Pull request by EvanLjp**
polish docs

### CODE REVIEW

**Comment by wu-sheng** [comment]
image

These 2 links are strange, should we remove them?

**Comment by EvanLjp** [comment]
> image
> 
> These 2 links are strange, should we remove them?

fixed

File: docs/en/guides/compile/compile.md [hunk 1]
```diff
@@ -1,10 +1,12 @@
 # Compile
```
Comment (wu-sheng) [inline docs/en/guides/compile/compile.md:1]: Compile -> Compiling

Don't use a verb in the document.

File: docs/en/guides/compile/compile.md [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
-## command
-```
-make build
-```
-## platform
-Linux and MacOs is supported in SkyWalking Satellite. Windows is not good supported beacuse some features is not adaptive on the Windows, such as the mmap feature. 
+
+## Platform
+Linux and MacOs is supported in SkyWalking Satellite. Windows is not good supported beacuse some features is not adaptive on the Windows, such as the mmap feature. If you want to run it on the windows platform, please read [the doc](../../FAQ/running_on_windows.md).
```
Comment (wu-sheng) [inline docs/en/guides/compile/compile.md:4]:
```suggestion
Linux and MacOS are supported in SkyWalking Satellite. Windows is not due to following components please read [the doc](../../FAQ/running_on_windows.md).
```

File: docs/en/guides/compile/compile.md [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
 
 The Windows platform does not support plugins list:
```
Comment (wu-sheng) [inline docs/en/guides/compile/compile.md:6]:
```suggestion
```

File: docs/en/guides/README.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# Guides
+If you want to debug or develop SkyWalking Satellite, The following documentations would guide you.
+
+- [How to contribute the plugins?](./contribuation/plugin.md)
+- [How to add unit test in the plugins?](./test/test.md)
+- [How to compile SkyWalking Satellite?](./compile/compile.md)
```
Comment (wu-sheng) [inline docs/en/guides/README.md:6]:
```suggestion
- [How to contribute a plugin?](./contribuation/plugin.md)
- [How to add unit test for a plugin?](./test/test.md)
- [How to compile SkyWalking Satellite?](./compile/compile.md)
```

**Review by wu-sheng — COMMENTED** [review]
1. Where could I find a recommend configuration in the source and dist?
2. In the root page, I even can't find the `setup` section? How should I use?
3. Is `plugin-description.md` doc auto generated? There are grammar issues, and the first letter of the sentences are not upper case. Could you check this part? And also, if it is auto generated, you should set up the CI process to make sure it is always updated in every commit.

File: docs/en/concepts-and-designs/overview.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+You can easily use it to connect to the SkyWalking ecosystem and enhance the capacity of SkyWalking. 
+There are some enhance features on the following when using Satellite.
+
+1. Provide a unified data collection layer to collect logs, traces, and metrics.
+2. Provide a safer local cache to reduce the memory cost of the service.
+3. Provide the unified transfer way shields the functional differences in the different language libs, such as MQ.
+4. Provides the preprocessing functions to ensure accuracy of the metrics, such as sampling.
+
+## Architecture
+SkyWalking Satellite is logically split into four parts: Gatherer, Processor, and Sender.
```
Comment (gxthrj) [inline docs/en/concepts-and-designs/overview.md:16]: `four parts` -> `three parts`

**Comment by EvanLjp** [comment]
> 1. Where could I find a recommend configuration in the source and dist?
That's configured in the plugin-description default config.

**Comment by wu-sheng** [comment]
> > 1. Where could I find a recommend configuration in the source and dist?
> >    That's configured in the plugin-description default config.

No, it isn't. The end users are focusing how to use `satellite_config.yaml` in the dist as step one, the documentation you provided represents nothing related to that, no hierarchy structure, key words.

Let me give an example, 
In your documentation, this is about fallback
image

This is how the YAML really looks like
image

How could a new users to link these two together? This is not easy even for me.

**Comment by wu-sheng** [comment]
You should have read OAP documentation, we need to talk about file format, then with a vocabulary to provide configuration in a clear structure. Otherwise, it is very confusing.

**Comment by EvanLjp** [comment]
> You should have read OAP documentation, we need to talk about file format, then with a vocabulary to provide configuration in a clear structure. Otherwise, it is very confusing.
I see what you mean.

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# Welcome
+**Here are SkyWalking Satellite official documentations. You're welcome to join us.**
+
+From here you can learn all about  **SkyWalking Satellite's** architecture, how to deploy and use SkyWalking Satellite.
+
+- [Concepts and Designs](en/concepts-and-designs/README.md). You'll find the most important core ideas about SkyWalking Satellite. You can learn from here if you want to understand what is going on under our cool features.
+- [Plugins description](en/plugins/plugin-description.md). You'll find the descriptions and the configurations of the whole plugins.
```
Comment (nic-chen) [inline docs/README.md:7]:
```suggestion
- [Plugins description](en/plugins/plugin-description.md). You'll find the descriptions and the configurations of all plugins. 
```

File: docs/README.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
+# Welcome
+**Here are SkyWalking Satellite official documentations. You're welcome to join us.**
+
+From here you can learn all about  **SkyWalking Satellite's** architecture, how to deploy and use SkyWalking Satellite.
+
+- [Concepts and Designs](en/concepts-and-designs/README.md). You'll find the most important core ideas about SkyWalking Satellite. You can learn from here if you want to understand what is going on under our cool features.
+- [Plugins description](en/plugins/plugin-description.md). You'll find the descriptions and the configurations of the whole plugins. 
+- [Guides](en/guides/README.md). You'll find how to develop or debug SkyWalking Satellite guides.
```
Comment (nic-chen) [inline docs/README.md:8]:
```suggestion
- [Guides](en/guides/README.md). You'll find guides about how to develop or debug SkyWalking Satellite.
```

File: docs/en/concepts-and-designs/project-goals.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+- **Light Weight**. SkyWalking Satellite has a limited cost for resources and high-performance because of the requirements of the sidecar deployment model.
+
+- **Pluggable**. SkyWalking Satellite core team provides many default implementations, but definitely it is not enough,
+and also don't fit every scenario. So, we provide a lot of features for being pluggable. 
+
+- **Portability**.  SkyWalking Satellite can run in multiple environments, including: 
+    - Use traditional deployment as a demon process to collect data.
+    - Use cloud services as a sidecar, such as in the kubernetes platform.
+
+- **Interop**.   Observability is a big landscape, SkyWalking is impossible to support all, even by its community. So Skywalking Satellite is compatible with many protocols, including:
```
Comment (kezhenxu94) [inline docs/en/concepts-and-designs/project-goals.md:13]:
```suggestion
- **Interop**.   Observability is a big landscape, SkyWalking is impossible to support all, even by its community. So SkyWalking Satellite is compatible with many protocols, including: 
```

File: README.md [hunk 9]
```diff
@@ -8,16 +8,22 @@ Apache SkyWalking Satellite
...
+Go to the [download page](https://skywalking.apache.org/downloads/) to download all available binaries, including MacOS and Linux.
+If you want to try the latest features or run on the Windows, however, you can compile the latest codes yourself, as the guide below. 
 
 # Compile
-[How to compile the Satellite.](./docs/compile/compile.md)
+As SkyWalking Satellite is using `Makefile`, compiling the project is as easy as executing a command in the root directory of the project.
+```shell script
+make build
+```
+Due to system compatibility problems, some plugins of SkyWalking Satellite cannot be used in Windows system. If you need to compile SkyWalking Satellite on Windows platform, please read [the doc](docs/en/guides/compile/compile.md).
```
Comment (wu-sheng) [inline README.md:26]: I can see `Windows` is your makefile, what do you mean ?

File: docs/en/setup/README.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
+# Setup
+First and most important thing is, SkyWalking Satellite startup behaviours are driven by configs/satellite_config.yaml. Understood the setting file will help you to read this document.
+
+## Startup script
+The startup script is /bin/skywalking-satellite-{version}-{plateform}-amd64. 
+1. Generate docs.
+```shell script
+./bin/skywalking-satellite-{version}-{plateform}-amd64 docs --output=./docs
+```
```
Comment (wu-sheng) [inline docs/en/setup/README.md:9]: Why need generate doc in the setup part?

Comment (EvanLjp) [reply docs/en/setup/README.md:9]: just descripe the supported command

File: README.md (hunk 9 above)
Comment (EvanLjp) [reply README.md:26]: platform means the supported platform，however，it not works. It only works when append windows  to the build

File: docs/en/setup/README.md (hunk 10 above)
Comment (EvanLjp) [reply docs/en/setup/README.md:9]: maybe should move it to the other part

File: docs/en/FAQ/README.md [hunk 11]
```diff
@@ -0,0 +1,3 @@
+## FAQ
+
+1. [How do we run Satellite on Windows?](./running_on_windows.md)
```
Comment (kezhenxu94) [inline docs/en/FAQ/README.md:3]: Having an individual doc for building Windows is troublesome, please review https://github.com/apache/skywalking-satellite/pull/24 and rework this part base on that

File: README.md (hunk 9 above)
Comment (EvanLjp) [reply README.md:30]: fixed

File: docs/en/guides/compile/compile.md [hunk 12]
```diff
@@ -1,7 +1,7 @@
 # Compiling
 
 ## Platform
-Linux and MacOS are supported in SkyWalking Satellite. Windows is not due to following components. Please read [the doc](../../FAQ/running_on_windows.md).
+Linux and MacOS are supported in SkyWalking Satellite. Windows is not due to following components.
```
Comment (kezhenxu94) [inline docs/en/guides/compile/compile.md:4]: Is Windows unsupported anyway, or do we have a fallback queue implementation on Windows? For example, users can try it on Windows with `memory-queue`?

Comment (EvanLjp) [reply docs/en/guides/compile/compile.md:4]: already push a memory queue

Comment (kezhenxu94) [reply docs/en/guides/compile/compile.md:4]: > already push a memory queue

I know that, I mean we cannot simply say "Windows is not due to following components.", `mmap-queue` is only a component in Satellite, users can still try / use Satellite if they configure another component (like `memory-queue`), the unavailability is component level and should be documented in that specific component, i.e. document "This component is not available on Windows" in [the `mmap-queue` plugin doc](https://github.com/apache/skywalking-satellite/blob/main/docs/plugins/queue/mmap/README.md), instead of here, "compile" doc.

Comment (wu-sheng) [reply docs/en/guides/compile/compile.md:4]: This is the question in my mind, if memory queue is available, then satellite should be available for WIndows platform. We should be clear about that. Could you deliver the window version binary? The current idea is saying code change, which is not a good one.

Comment (EvanLjp) [reply docs/en/guides/compile/compile.md]: fixed

File: README.md [hunk 13]
```diff
@@ -8,16 +8,34 @@ Apache SkyWalking Satellite
...
-- [The first design of Satellite 0.1.0](https://skywalking.apache.org/blog/2020-11-25-skywalking-satellite-0.1.0-design/)
-- [The project structure](./docs/project_structue.md)
+- [Official documentation](https://skywalking.apache.org/docs/)
+- [Blog](https://skywalking.apache.org/blog/2020-11-25-skywalking-satellite-0.1.0-design/) about the design of Satellite 0.1.0.
+
+NOTICE, SkyWalking Satellite uses [v3 protocols](https://github.com/apache/skywalking/blob/master/docs/en/protocols/README.md). They are incompatible with previous SkyWalking releases before SkyWalking 8.0.
 
 # Download
+Go to the [download page](https://skywalking.apache.org/downloads/) to download all available binaries, including MacOS and Linux.
+If you want to try the latest features or run on the Windows, however, you can compile the latest codes yourself, as the guide below.
```
Comment (kezhenxu94) [inline README.md:19]: Should be reworded, we will deliver a Windows binary, right?

File: README.md [hunk 14]
```diff
@@ -8,16 +8,34 @@ Apache SkyWalking Satellite
...
-[How to compile the Satellite.](./docs/compile/compile.md)
+As SkyWalking Satellite is using `Makefile`, compiling the project is as easy as executing a command in the root directory of the project.
+```shell script
+git clone https://github.com/apache/skywalking-satellite
+cd skywalking-satellite
+git submodule init
+git submodule update
+make build
+```
+Due to system compatibility problems, some plugins of SkyWalking Satellite cannot be used in Windows system. If you need to compile SkyWalking Satellite on Windows platform, please read [the doc](docs/en/guides/compile/compile.md).
```
Comment (kezhenxu94) [inline README.md:30]:
```suggestion
Due to system compatibility problems, some plugins of SkyWalking Satellite cannot be used in Windows system. Check the corresponding documentation to see whether the plugin is available on Windows.
```

File: docs/en/setup/plugins/queue_mmap-queue.md [hunk 15]
```diff
@@ -0,0 +1,20 @@
+# Queue/mmap-queue
+## Description
+this is a memory mapped queue to provide the persistent storage.
+## DefaultConfig
```
Comment (kezhenxu94) [inline docs/en/setup/plugins/queue_mmap-queue.md:4]: You should indicate the Windows compatibility here

File: docs/en/setup/plugins/queue_mmap-queue.md [hunk 16]
```diff
@@ -0,0 +1,20 @@
+# Queue/mmap-queue
+## Description
+this is a memory mapped queue to provide the persistent storage.
```
Comment (kezhenxu94) [inline docs/en/setup/plugins/queue_mmap-queue.md:3]: The description (and other plugins') should be capitalized.

File: docs/en/guides/compile/compile.md (hunk 12 above)
Comment (EvanLjp) [reply docs/en/guides/compile/compile.md]: zhenxu help me to unify the compile at all platforms. Thanks a lot.

File: docs/en/concepts-and-designs/README.md [hunk 17]
```diff
@@ -0,0 +1,13 @@
+# Concepts and Designs
+Concepts and Designs help you to learn and understand the SkyWalking Satellite and the landscape.
+
+- What is SkyWalking Satellite?
+  - [Overview and Core concepts](overview.md). Provides a high-level description and introduction, including the problems the project solves.
+  - [Project Goals](project-goals.md). Provides the goals, which SkyWalking Satellite is trying to focus and provide features about them.
+
+After you read the above documents, you should understand the SkyWalking basic goals. Now, you can choose which following parts
```
Comment (nic-chen) [inline docs/en/concepts-and-designs/README.md:8]:
```suggestion
After you read the above documents, you should understand basic goals of the SkyWalking Satellite. Now, you can choose which following parts 
```

File: docs/en/concepts-and-designs/overview.md [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+3. Provide the unified transfer way shields the functional differences in the different language libs, such as MQ.
+4. Provides the preprocessing functions to ensure accuracy of the metrics, such as sampling.
+
+## Architecture
+SkyWalking Satellite is logically split into three parts: Gatherer, Processor, and Sender.
+
+<img src="https://skywalking.apache.org/blog/2020-11-25-skywalking-satellite-0.1.0-design/Satellite.png"/>
+
+- Gatherer collect data and reformat them for SkyWalking requirements.
+- Processor processes the input data to generate the new data for  Observability.
```
Comment (nic-chen) [inline docs/en/concepts-and-designs/overview.md:21]:
```suggestion
- Processor processes the input data to generate the new data for Observability.
```

File: docs/en/concepts-and-designs/project-goals.md [hunk 19]
```diff
@@ -0,0 +1,16 @@
+# Design Goals
+The document outlines the core design goals for SkyWalking Satellite project.
+
+- **Light Weight**. SkyWalking Satellite has a limited cost for resources and high-performance because of the requirements of the sidecar deployment model.
+
+- **Pluggable**. SkyWalking Satellite core team provides many default implementations, but definitely it is not enough,
+and also don't fit every scenario. So, we provide a lot of features for being pluggable. 
+
+- **Portability**.  SkyWalking Satellite can run in multiple environments, including:
```
Comment (nic-chen) [inline docs/en/concepts-and-designs/project-goals.md:9]:
```suggestion
- **Portability**. SkyWalking Satellite can run in multiple environments, including: 
```

File: docs/en/concepts-and-designs/project-goals.md [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+- **Light Weight**. SkyWalking Satellite has a limited cost for resources and high-performance because of the requirements of the sidecar deployment model.
+
+- **Pluggable**. SkyWalking Satellite core team provides many default implementations, but definitely it is not enough,
+and also don't fit every scenario. So, we provide a lot of features for being pluggable. 
+
+- **Portability**.  SkyWalking Satellite can run in multiple environments, including: 
+    - Use traditional deployment as a demon process to collect data.
+    - Use cloud services as a sidecar, such as in the kubernetes platform.
+
+- **Interop**.   Observability is a big landscape, SkyWalking is impossible to support all, even by its community. So SkyWalking Satellite is compatible with many protocols, including:
```
Comment (nic-chen) [inline docs/en/concepts-and-designs/project-goals.md:13]:
```suggestion
- **Interop**.  Observability is a big landscape, SkyWalking is impossible to support all, even by its community. So SkyWalking Satellite is compatible with many protocols, including: 
```

**Comment by wu-sheng** [comment]
Please add MacOS and Windows platform in the CI processes, https://github.com/apache/skywalking-satellite/blob/main/.github/workflows/build-and-test.yaml#L27

File: docs/en/setup/README.md [hunk 21]
```diff
@@ -0,0 +1,19 @@
+# Setup
+First and most important thing is, SkyWalking Satellite startup behaviours are driven by configs/satellite_config.yaml. Understood the setting file will help you to read this document.
+
+## Startup script
+The startup script is /bin/skywalking-satellite-{version}-{plateform}-amd64. 
+1. Start SkyWalking Satellite.
+```shell script
+./bin/skywalking-satellite-{version}-{plateform}-amd64 start --config=./configs/satellite_config.yaml
```
Comment (wu-sheng) [inline docs/en/setup/README.md:8]: This is correct and not correct at the same time. This actually is a dev-command for the source code local compiling startup.
But usually, like you see in SkyWalking's main repo release, you should have a DIST tar, including binaries and config file in the same folder. 
@EvanLjp Do you plan to use another PR to do this?

File: docs/en/concepts-and-designs/project_structue.md [hunk 22]
```diff
@@ -0,0 +1,33 @@
+# Project Structure
+- cmd: The starter of Satellite.
+- configs: Satellite configs.
+- internal: Core, Api, and common utils.
```
Comment (wu-sheng) [inline docs/en/concepts-and-designs/project_structue.md:4]:
```suggestion
- internal: Core, API, and common utils.
```

**Comment by wu-sheng** [comment]
You should run `generate_plugin_doc.go` in the CI, and compare the diff with sources.

**Comment by wu-sheng** [comment]
Others generally good to me. The documentation is better now.

File: docs/en/setup/README.md (hunk 21 above)
Comment (EvanLjp) [reply docs/en/setup/README.md:8]: please ignore it, would add the start script in another PR for release.

**Review by wu-sheng — APPROVED** [review]
To all reviewers, I am going to merge this. We should polish documents continuously.
